Feature/credential component

BaskarMitrah opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

dc297 commented on packages/gatsby-theme-aio/gatsby-config.js:
```diff
@@ -154,4 +156,14 @@ module.exports = {
       },
     },
   ],
+  developMiddleware: app => {
+    app.use(
+      "/console/api",
+      createProxyMiddleware({
+        target: "https://developer-stage.adobe.com",
+        secure: false,
+        changeOrigin: true,
+      })
+    )
+  },
```
Do we need this? Since dev site is hosted on the same domain as dev console, ideally this should only be needed for local development 🤔

louisachu commented on packages/gatsby-theme-aio/src/components/GetCredential/FormFields.js:
```diff
@@ -0,0 +1,189 @@
+  });
+}
+
+export const downloadAndModifyZip = async (zipFileURL) => {
```
The download is happening even if i don't selected the download check box.

louisachu reviewed (commented):
Overall, please check all the href links that the experience should always open a new window.  Right now, most of the links are just loading within the page.

louisachu commented on example/src/pages/credential/Credential.css:
```diff
@@ -0,0 +1,17 @@
+.conatiner {
```
Is this only needed if the author likes to change things around?  By default, this shouldn't be needed, correct?

BaskarMitrah replied:
Yes it's based on the user need

BaskarMitrah commented on example/src/pages/credential/GetCredentialExternal.js:
```diff
@@ -0,0 +1,60 @@
+
+      <GetCredential.UnknownError helpLink="https://some_help_link" helpLinkText="Get Help" className="unKnownError" />
+
+      <GetCredential.Card title="Your credential is ready to use" paragraph="Check the downloads section of your browser for the ZIP file, or find it where you save downloads on your machine." nextStepsLabel="Next steps" nextStepsHref="/credentials/nextsteps" devConsoleDirection="project_overview|api_overview|credential_overview" developerConsoleManage="Manage on Developer console" className="car ...
```
Once it's moved to the staging it works it depends on the what user will provide

dc297 replied:
Hi @BaskarMitrah let's wire this up to the project overview page for not. We can set the href to the following
```
/console/projects/{orgId}/{projectId}/overview
```

where `orgId` is the org id of the selected org and `projectId` is received in the response from the create credential endpoint. Please let me know if you have any questions.

dc297 commented:
@BaskarMitrah found a few more bugs:
1. You should not be allowed to submit the form if the name is invalid. Steps to reproduce:
	a. Enter "test" in credential name
	b. Enter [www.adobe.com](http://www.adobe.com/) in allowed domain
	c. Select download checkbox
	d. Select terms checkbox
	e. The create credential button gets enabled and the user is allowed to submit the form. Instead the create credential button should remain disabled because the credential name is not valid
2. Changing organization popup does not show the selected org
	a. Click the change organization button
	b. From the popup select an org different than the one selected by default
	c. Click save
	d. Click the change organization button again
	e. Notice that the default organization is selected instead the current selected org should be selected by default
3. Noticed an error on the browser's console
	
	![image](https://github.com/adobe/aio-theme/assets/3163579/e1da69da-86dc-4028-a0e7-bc0d56d6e1cb)

4. Not sure why but the download API is getting called twice
	
	![image](https://github.com/adobe/aio-theme/assets/3163579/4b654fe3-4ae1-4953-a5d1-81d113dc9485)

BaskarMitrah commented:
> @BaskarMitrah found a few more bugs:
> 
> 1. You should not be allowed to submit the form if the name is invalid. Steps to reproduce:
>    a. Enter "test" in credential name
>    b. Enter [www.adobe.com](http://www.adobe.com/) in allowed domain
>    c. Select download checkbox
>    d. Select terms checkbox
>    e. The create credential button gets enabled and the user is allowed to submit the form. Instead the create credential button should remain disabled because the credential name is not valid
> 2. Changing organization popup does not show the selected org (same as what @louisachu reported [here](https://github.com/adobe/aio-theme/pull/1536#discussion_r1326449929))
>    a. Click the change organization button
>    b. From the popup select an org different than the one selected by default
>    c. Click save
>    d. Click the change organization button again
>    e. Notice that the default organization is selected instead the current selected org should be selected by default
> 3. Noticed an error on the browser's console
>    ![image](https://user-images.githubusercontent.com/3163579/268349303-e1da69da-86dc-4028-a0e7-bc0d56d6e1cb.png)
> 4. Not sure why but the download API is getting called twice
>    ![image](https://user-images.githubusercontent.com/3163579/268349346-4b654fe3-4ae1-4953-a5d1-81d113dc9485.png)
> 5. There should be no return experience. Steps to reproduce:
>    a. Submit the form to successfully create a credential
>    b. Refresh the page or open a new window. The component stays on the card view
>    ![Screenshot 2023-09-15 at 10 25 57 AM](https://user-images.githubusercontent.com/3163579/268352843-e3c24288-317f-4d6d-a817-3fc2de5175fb.png)
>    c. Instead, the component should switch back to the form
> 6. The credential is always created in the same org. Steps to reproduce:
>    a. Use the change organization link to change org to something other than default
>    b. Fill the form and create the credential
>    c. The credential gets created in the default org not the one selected by the user. Also, The card view shows the default org not the one selected by the user.
> 7. The `Go to Developer Console` link below the form should point to `/console` instead of `https://developer.adobe.com/console/`
> 8. I see an error boundary has not been setup for the component as well. If you would like, we can get on a call and discuss how to do that.
> 9. Zip download does not work. Steps to reproduce:
>    a. I updated the download option in `GetCredentialExternal.js` to the following:
>    ```
>    <GetCredential.Form.Download title="JavaScript" href="https://acrobatservices.adobe.com/dc-integration-creation-app-cdn/8bab684/files/samples_q3_2023/PROD/dc-pdf-services-sdk-java-samples.zip" />
>    ```
>    
>    
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    Notice I have added a valid url.
>    b. After submitting the form, I just get a zip file with name JavaScript.zip containing only credential.json. Instead, the zip file from the url should be downloaded with the credential.json injected.
> 10. The call to `/console/api/organizations` is being made every time the component renders.

1) Completed
2) Done
3)Not Showing now
4)Fixed 
5)Fixed
6)Fixed
7)Fixed
8)Still Needs to be implemented
9)Fixed
10) Fixed

louisachu reviewed (commented):
Please fix all the links to be target to a blank tab instead of loading on the page.

BaskarMitrah commented:
> @BaskarMitrah Thanks for fixing the bugs 🙌 . I just did another round of testing and looks like all the bugs are fixed except 7. It should be a quick fix though, I'll prepare a fix.
> 
> While switching orgs, I noticed a strange issue while switching to `IO Team Org - Stage`. I've shared a video with you on Slack.
> 
> I also see some css issue like on the side component:
> 
> ![Screenshot 2023-09-19 at 10 42 13 AM](https://user-images.githubusercontent.com/3163579/269047918-3d0a02c2-192e-408e-ad18-f1791369324f.png)

I've fixed the org changing issue

BaskarMitrah commented:
> Please fix all the links to be target to a blank tab instead of loading on the page.

Done
